File: university-service/src/main/java/com/university/service/ResultService.java
package com.university.service;

import com.university.dto.ResultReqDTO;
import com.university.dto.ResultRespDTO;
import com.university.dto.SemesterResultRespDTO;
import java.util.List;

public interface ResultService {
    ResultRespDTO uploadResult(ResultReqDTO dto);
    ResultRespDTO updateResult(Long id, ResultReqDTO dto);
    ResultRespDTO getResultById(Long id);
    List<ResultRespDTO> getAllResults();
    List<ResultRespDTO> getResultsByStudent(Long studentId);
    SemesterResultRespDTO getSemesterResult(Long studentId, Long semesterId);
    void deleteResult(Long id);
}

File: university-frontend/university-frontend/src/services/resultService.js
import axiosInstance from '../api/axiosInstance';


// Fetch all results records
export const getAllResults = () => axiosInstance.get('/results');

// Fetch a specific resultsbystudent by ID
export const getResultsByStudent = (studentId) => axiosInstance.get(`/results/student/${studentId}`);

// Fetch a specific semesterresult by ID
export const getSemesterResult = (studentId, semesterId) =>
  axiosInstance.get(`/results/student/${studentId}/semester/${semesterId}`);
export const uploadResult = (data) => axiosInstance.post('/results', data);

// Update an existing result record
export const updateResult = (id, data) => axiosInstance.put(`/results/${id}`, data);

// Delete a result record
export const deleteResult = (id) => axiosInstance.delete(`/results/${id}`);
